rnd/autogpt_server/autogpt_server/cli.py

```diff
@@ -0,0 +1,97 @@
+from multiprocessing import freeze_support
+from multiprocessing.spawn import freeze_support as freeze_support_spawn
+
+import click
+
+
+@click.group()
+def main():
+    """AutoGPT Server CLI Tool"""
+    pass
+
+
+@main.command()
+def background() -> None:
+    from autogpt_server.app import background_process
+
+    background_process()
+
+
+@main.command()
+def run():
+    import os
+    import pathlib
+    import subprocess
+
+    # Define the path for the new directory and file
+    home_dir = pathlib.Path.home()
+    new_dir = home_dir / ".config" / "agpt"
+    file_path = new_dir / "running.tmp"
+
+    # Create the directory if it does not exist
+    os.makedirs(new_dir, exist_ok=True)
+    if file_path.exists():
+        print("Server is already running")
+        return
+
+    sp = subprocess.Popen(
+        ["poetry", "run", "python", "autogpt_server/cli.py", "background"],
+        stdout=subprocess.DEVNULL,  # Redirect standard output to devnull
+        stderr=subprocess.DEVNULL,  # Redirect standard error to devnull
+    )
+    print(f"Server running in process: {sp.pid}")
+
+    with open(file_path, "w") as file:
```

Comment:
Finally block to remove?  otherwise we might be having dangling running.tmp but the process is already dead.

Btw would it be simpler to use `lsof` for this instead of file redundancy?

Reply:
Yeah thats how I did it for the autogpt cli, which I could look at doing here. 

However, that comes with the edge case of needing to know the port the process is using. Which makes things complicated if we all users to choose the port.

I was thinking further now the line of have a process table in the database and updating records in that. Doing so would allow us to handle all these edge cases